kafka-client/src/main/java/dev/responsive/kafka/store/ResponsivePartitionedStore.java

```diff
@@ -169,7 +169,7 @@ public boolean persistent() {
 
   @Override
   public void put(final Bytes key, final byte[] value) {
-    buffer.put(key, value);
+    buffer.put(key, value, context.timestamp());
```

Comment:
cc @ableegoldman - is this timestamp valid? this way we won't rely on the timestamp KV store functionality at all

Reply:
Ah...sort of? I believe this is the timestamp of the record that was read from the input topic, which is probably good enough but definitely not strictly-speaking correct. 

Some additional context in this KIP (the relevance being that we used to use this method to get a timestamp when one was not provided to a `WindowStore#put` API, but we deprecated/removed  that API entirely because this wasn't semantically sound -- https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=115526545

Reply:
Like I said though, for our purposes this is probably ok. If someone ever complains about it we can consider extracting it from the `TimestampAndValue` for timestamped stores to improve the semantics

The one thing I'm not certain of is what happens if you try to invoke this from a punctuator (where there is no "input record"). I'd recommend writing a quick test for this just to make sure it doesn't throw an exception or return complete garbage

Reply:
If it helps, I just checked the code and this is what we use for the changelog timestamp, so it's actually probably more correct than the "real" message timestamp (for our purposes)